src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -72,4 +78,25 @@ public static boolean matchesKeys(ResourceLocation loc, RegistryKey<?>... keys)
 				return true;
 		return false;
 	}
-}
\ No newline at end of file
+
+	/**
+	 * <p>Registers a {@link IDispenseItemBehavior} that will perform the new behavior if the condition is met and the behavior that was already in the registry if not.</p><br>
+	 * This works even if multiple mods add new behavior to the same item.
+	 *
+	 * @param item the {@link Item} to register the {@code newBehavior} for.
+	 * @param condition a {@link BiPredicate} that takes {@link IBlockSource} and {@link ItemStack} arguments,
+	 *                  returning true if the {@code newBehavior} should be performed. Ideally, this should be implemented
+	 *                  such that the predicate only passes if the new behavior will be 'successful,' avoiding
+	 *                  problems with failure sounds.
```

Comment:
I feel like this is way too much documentation for one parameter. It'd be cleaner to explain this in the main method doc.

Reply:
Changed in latest commit.